rotkehlchen/chain/evm/decoding/llamazip/decoder.py

```diff
@@ -0,0 +1,89 @@
+import abc
+import logging
+from collections.abc import Callable
+from typing import TYPE_CHECKING
+
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.llamazip.constants import CPT_LLAMAZIP, LLAMAZIP_CPT_DETAILS
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.evm.decoding.utils import maybe_reshuffle_events
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress, EvmTransaction
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class LlamazipCommonDecoder(DecoderInterface, abc.ABC):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            router_addresses: tuple['ChecksumEvmAddress', ...],
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.router_addresses = router_addresses
+
+    def _decode_swap(
+            self,
+            transaction: EvmTransaction,
+            decoded_events: list['EvmEvent'],
+            all_logs: list['EvmTxReceiptLog'],  # pylint: disable=unused-argument
+    ) -> list['EvmEvent']:
+        spend_event = receive_event = None
+        for event in decoded_events:
+            if (
+                (event.event_type == HistoryEventType.TRADE and event.event_subtype == HistoryEventSubType.SPEND) or  # noqa: E501
+                (event.event_type == HistoryEventType.SPEND and event.event_subtype == HistoryEventSubType.NONE)  # noqa: E501
+            ):
+                spend_event = event
+                spend_event.counterparty = CPT_LLAMAZIP
+                spend_event.event_type = HistoryEventType.TRADE
+                spend_event.event_subtype = HistoryEventSubType.SPEND
+                spend_event.notes = f'Swap {event.balance.amount} {event.asset.symbol_or_name()} in LlamaZip'  # noqa: E501
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
```

Comment:
why not trade/receive also here? ARe you sure this can't happen?

Reply:
It is always the uniswap router what gets used so this case shouldn't appear unless something in our logic changes there